ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -1220,6 +1221,52 @@ def check_sonic_version_after_reboot(self):
                 raise Exception("Sonic upgrade failed. Target={} Current={}".format(\
                     target_version, current_version))
 
+    def check_reboot_cause_history_after_reboot(self):
+        """
+        The function is to check the reboot hihstory after reboot
+        Reboot history example:
+            show reboot-cause history
+            Name                 Cause                                Time                             User    Comment
+            -------------------  -----------------------------------  -------------------------------  ------  ---------
+            2021_09_02_22_18_31  warm-reboot                          Thu 02 Sep 2021 10:17:34 PM UTC  admin   N/A
+            2021_09_02_20_03_01  reboot                               Thu 02 Sep 2021 08:02:04 PM UTC  admin   N/A
+            2021_09_02_19_53_36  Hardware - Other (Reset from ComEx)  N/A                              N/A     N/A
+        """
+        reboot_history_pattern = {
+            "title": r"^Name[\s]{2,}Cause[\s]*Time[\s]{2,}User[\s]{2,}Comment",
+            "split_dotted_line": r"^-------------------[\s]{2}[-]*[\s]{2}[-]*[\s]{2}[-]*[\s]{2}[-]*",
+            "reboot_cause_content": r"^\d{4}_(0[1-9]|1[0-2])_(0[1-9]|[1-2][0-9]|3[0-1])_(20|21|22|23|[0-1]\d)_"
+                                    r"[0-5]\d_[0-5]\d[\s]{2,}(reboot|fast-reboot|warm-reboot|Watchdog .*|Unknown .*|"
+                                    r"Hardware .*|Power Loss .*)[\s]{2,}(N\/A|(Mon|Tue|Wed|Thu|Fri|Sat|Sun) .*)"
+                                    r"[\s]{2,}(.*)[\s]{2,}(.*)"
+        }
+        stdout, stderr, return_code = self.dut_connection.execCommand("show reboot-cause history")
```

Comment:
The above command output makes it suitable for use show_and_parse: https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices/sonic.py#L1139

Reply:
This is a PTF script. The `show_and_parse` method is not available here.